bciers/apps/reporting/src/app/components/operations/OperationReview.tsx

```diff
@@ -1,144 +1,197 @@
+      ...formDataObject,
+      activities: formDataObject.activities?.map((activityId: any) => {
+        const activityName = allActivities.find((a) => a.id === activityId)
+          ?.name;
+        return activityName || activityId;
+      }),
+      regulated_products: formDataObject.regulated_products?.map(
+        (productId: any) => {
+          const productName = allRegulatedProducts.find(
+            (p) => p.id === productId,
+          )?.name;
+          return productName || productId;
+        },
+      ),
+    };
+
+    try {
+      await actionHandler(endpoint, method, pathToRevalidate, {
+        body: JSON.stringify(preparedData),
+        headers: {
+          "Content-Type": "application/json",
+          accept: "application/json",
+        },
+      });
+      router.push(saveAndContinueUrl);
+    } catch (error) {
+      console.error("Submission failed:", error);
+    }
+  };
+
+  useEffect(() => {
+    if (!formData || !allActivities || !allRegulatedProducts) {
+      return;
+    }
+
+    const activities = formData.activities || [];
+    const products = formData.regulated_products || [];
+
+    const preselectedActivities = activities;
+    const preselectedProducts = products;
+
+    setSchema((prevSchema) => ({
+      ...prevSchema,
+      properties: {
+        ...prevSchema.properties,
+        activities: {
+          type: "array",
+          title: "Reporting activities",
+          items: {
+            type: "number",
+            enum: allActivities.map((activity) => activity.id),
+            enumNames: allActivities.map((activity) => activity.name),
+          },
+          uniqueItems: true,
+        },
+        regulated_products: {
+          type: "array",
+          title: "Regulated products",
+          items: {
+            type: "number",
+            enum: allRegulatedProducts.map((product) => product.id),
+            enumNames: allRegulatedProducts.map((product) => product.name),
+          },
+          uniqueItems: true,
+        },
+        operation_representative_name: {
+          type: "string",
+          title: "Operation representative",
+          enum: [formData.operation_representative_name || ""],
+        },
+        operation_type: {
+          type: "string",
+          title: "Operation type",
+          enum: [formData.operation_type || ""],
+        },
+        date_info: {
+          type: "object",
+          readOnly: true,
+          title: `Please ensure this information was accurate for ${reportingWindowEnd}`,
+        },
+      },
+    }));
+
+    const updatedFormData = {
+      ...formData,
+      activities: preselectedActivities,
+      regulated_products: preselectedProducts,
+    };
+
+    setFormDataState(updatedFormData);
+    setLoading(false);
+  }, [allActivities, allRegulatedProducts, formData]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  if (!formData || formData.error) {
+    return <div>No version ID found(TBD)</div>;
+  }
 
```

Comment:
Same here, I would let the generic react error boundary deal with this case. Otherwise we swallow an error and instead render something that may or may not be the root cause of the error